nebula2/sclient/GraphStorageClient.py

```diff
@@ -58,14 +58,15 @@ def _create_connection(self):
         storage_addrs = self._meta_cache.get_all_storage_addrs()
         if len(storage_addrs) == 0:
             raise RuntimeError('Get storage address from meta cache is empty')
-        try:
-            for addr in storage_addrs:
+        for addr in storage_addrs:
+            try:
                 conn = GraphStorageConnection(addr, self._time_out, self._meta_cache)
                 conn.open()
                 self._connections.append(conn)
-        except Exception as e:
-            logging.error('Create storage connection failed: {}'.format(e))
-            raise
+            except Exception as e:
+                logging.warning('Create storage connection to {} failed: {}'.format(addr, e))
+        if len(self._connections) == 0:
```

Comment:
I don't think this is reasonable. The StorageClient obtained from Metaclient should be reasonable. If it is a problem of the server, it means that the user has configured IP in this way before, and the client will report an error. Instead of ignoring these IPs.

Reply:
well, I think you are right when you are using a self-deployed cluster or have full permission to modify the cluster's configuration. But a python sdk user most likely doesn't have such permission.

Reply:
Actually, I'm also confused since I deployed this cluster using docker-compose.yaml from https://github.com/vesoft-inc/nebula-docker-compose on branch v2.0.0

Reply:
> well, I think you are right when you are using a self-deployed cluster or have full permission to modify the cluster's configuration. But a python sdk user most likely doesn't have such permission.

Clients can't ignore any IP information they get from the Meta. You can modify the `get_all_storage_addrs` to ensure that all hosts are valid, such as the offline  hosts

Reply:
ok, that makes sense I think. But I need some time since I have only used this for several days. BTW, is `mclient.__init__.MetaClient.list_hosts` the method I should focus on?

Reply:
yes, the return value `HostItem` with the host status.